core/src/main/scala/com/nvidia/spark/rapids/tool/GpuDevice.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids.tool
+
+import com.nvidia.spark.rapids.tool.profiling.ClusterProperties
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.rapids.tool.GpuTypes
+import org.apache.spark.sql.rapids.tool.util.StringUtils
+
+/**
+ * Abstract class representing a GPU device
+ */
+abstract class GpuDevice {
+  def getMemory: String
+  def getAdvisoryPartitionSizeInBytes: Option[String] = None
+  def getInitialPartitionNum: Option[Int] = None
```

Comment:
same here, maybe the abstract method returns the default as the original behavior?

Reply:
Addressed in previous comment.